CollectionOfHelpers/CollectionOfHelpers/Specialised/Sudoku/SudokuDistributedGridCollection.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace CollectionOfHelpers.Specialised.Sudoku
+{
+    public class SudokuDistributedGridCollection
+    {
+        public int GridSize => Columns.Length;
+        private int boxSize;
+
+        private SudokuSubCollection[] Columns;
+        private SudokuSubCollection[] Rows;
+        private SudokuSubCollection[] Boxes;
+
+        public SudokuDistributedGridCollection(int gridSize)
+        {
+            throwExceptionForInvalidGridSize(gridSize);
+            boxSize = (int) Math.Sqrt(gridSize);
+            initialiseGrid(gridSize);
+        }
+
+        private void throwExceptionForInvalidGridSize(int gridSize)
```

Comment:
In my pluralsight videos they preferred to name this sort of method "ValidateGridSize". Not sure if it's standard practice to include 'throwException' in the name since in future the code could be written to have a default value instead of throw an exception